Feature/kms 2279

anuradhas248 opened the pull request:
Pull Request for KMS-2279 Story: OS Tree View - Expand display in tree view to accommodate multiple layers for readability

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/opensalt/opensalt/263)
<!-- Reviewable:end -->

roverwolf commented:
The pull request needs to have the latest updates from the develop branch merged in and the conflicting file resolved.

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[app/Resources/assets/js/lsdoc/index.js, line 427 at r2](https://reviewable.io:443/reviews/opensalt/opensalt/263#-KzZoJDp5b3mRvbPlpdv:-KzZoJDp5b3mRvbPlpdw:ba917hi) ([raw file](https://github.com/opensalt/opensalt/blob/a657d1aabe705c38aff0e1edf3fd0dfc3edcbf40/app/Resources/assets/js/lsdoc/index.js#L427)):*
> ```JavaScript
> 
> function toggleDivRight() {
>     var RightWindow = document.getElementById("treeSideRight");
> ```

For case consistency, we should use "rightWindow" and "leftWindow"

---

*[app/Resources/assets/js/lsdoc/index.js, line 428 at r2](https://reviewable.io:443/reviews/opensalt/opensalt/263#-KzZoQTBgcmnUmsaW_nx:-KzZoQTBgcmnUmsaW_ny:bwxnp18) ([raw file](https://github.com/opensalt/opensalt/blob/a657d1aabe705c38aff0e1edf3fd0dfc3edcbf40/app/Resources/assets/js/lsdoc/index.js#L428)):*
> ```JavaScript
> function toggleDivRight() {
>     var RightWindow = document.getElementById("treeSideRight");
>    
> ```

3 unneeded spaces left on the line here.

---

*[app/Resources/assets/js/lsdoc/index.js, line 438 at r2](https://reviewable.io:443/reviews/opensalt/opensalt/263#-KzZoX1wWAh1aKj3ONRS:-KzZoX1wWAh1aKj3ONRT:b-lvvr8u) ([raw file](https://github.com/opensalt/opensalt/blob/a657d1aabe705c38aff0e1edf3fd0dfc3edcbf40/app/Resources/assets/js/lsdoc/index.js#L438)):*
> ```JavaScript
>         document.getElementById("treeSideLeft").setAttribute("style","width:100%");
>         document.getElementById("toggleRight").setAttribute("class", "fa fa-chevron-circle-left");
>            }
> ```

Should line up with the if and else lines.

---

*[app/Resources/assets/js/lsdoc/index.js, line 630 at r2](https://reviewable.io:443/reviews/opensalt/opensalt/263#-KzZpQTsfy2F0EFutaOe:-KzZpQTsfy2F0EFutaOf:b8ecla5) ([raw file](https://github.com/opensalt/opensalt/blob/a657d1aabe705c38aff0e1edf3fd0dfc3edcbf40/app/Resources/assets/js/lsdoc/index.js#L630)):*
> ```JavaScript
>     if (dragging)
>     {
>         $('#treeSideLeft').css("width",e.pageX+2);
> ```

For consistency we should probably use either `.css("width", ...)` or `.width(...)`, but both are used in this block of code for the same thing.

It would probably be better to check against the max width first and then set the width instead of setting it and then setting it again if it exceeds to avoid the user from seeing the second resize.

---

*[app/Resources/assets/js/lsdoc/index.js, line 637 at r2](https://reviewable.io:443/reviews/opensalt/opensalt/263#-KzZoglgvxbZJiEv-v0m:-KzZoglgvxbZJiEv-v0n:b-z9r14) ([raw file](https://github.com/opensalt/opensalt/blob/a657d1aabe705c38aff0e1edf3fd0dfc3edcbf40/app/Resources/assets/js/lsdoc/index.js#L637)):*
> ```JavaScript
>         }
>         $('#treeSideRight').width($("#treeView").width() - $("#treeSideLeft").width());
>         $(document).unbind('mousemove');
> ```

What `mousemove` is being unbound here?

If it is needed, should use `.off()` instead as `.unbind()` has been deprecated for a while.

---

*[app/Resources/assets/sass/cftree/view.scss, line 12 at r2](https://reviewable.io:443/reviews/opensalt/opensalt/263#-KzZq6vChOefr_EEPMGW:-KzZq6vDXsYZPNGCDgYt:brq9ei6) ([raw file](https://github.com/opensalt/opensalt/blob/a657d1aabe705c38aff0e1edf3fd0dfc3edcbf40/app/Resources/assets/sass/cftree/view.scss#L12)):*
> ```SCSS
> @media (min-width:330px) {
>     #ls_doc_list_left label.col-sm-2.control-label, #ls_doc_list_right label.col-sm-2.control-label {
> width:18%;
> ```

Should be indented.

---

*[app/Resources/assets/sass/cftree/view.scss, line 25 at r2](https://reviewable.io:443/reviews/opensalt/opensalt/263#-KzZq9qisJQJEJ-s4AOr:-KzZq9qisJQJEJ-s4AOs:b-astjps) ([raw file](https://github.com/opensalt/opensalt/blob/a657d1aabe705c38aff0e1edf3fd0dfc3edcbf40/app/Resources/assets/sass/cftree/view.scss#L25)):*
> ```SCSS
> #treeSideLeft {
>    // border-right:3px solid #f5f5f5;
>     Float:left;
> ```

`Float` should be all lower-case to be consistent.

---

*[app/Resources/assets/sass/cftree/view.scss, line 74 at r2](https://reviewable.io:443/reviews/opensalt/opensalt/263#-KzZqNtfRBwTGYLSe-lc:-KzZqNtfRBwTGYLSe-ld:b-e4uthx) ([raw file](https://github.com/opensalt/opensalt/blob/a657d1aabe705c38aff0e1edf3fd0dfc3edcbf40/app/Resources/assets/sass/cftree/view.scss#L74)):*
> ```SCSS
>     width:calc(100% - 40px);;
>     float:left;
>   padding-left: 5px;
> ```

style should be aligned here with the other styles, same with a few places below.

---

*[src/CftfBundle/Resources/views/DocTree/view.html.twig, line 113 at r2](https://reviewable.io:443/reviews/opensalt/opensalt/263#-KzZrSZOf4eNnFfg2Ijv:-KzZrSZOf4eNnFfg2Ijw:b-y51azs) ([raw file](https://github.com/opensalt/opensalt/blob/a657d1aabe705c38aff0e1edf3fd0dfc3edcbf40/src/CftfBundle/Resources/views/DocTree/view.html.twig#L113)):*
> ```Twig
>         </div>
>         <div id="dragbar"></div>
> ```

Lots of unneeded blank lines here, one should be enough if we want to separate the parts of the content.

---

*[src/CftfBundle/Resources/views/DocTree/view.html.twig, line 122 at r2](https://reviewable.io:443/reviews/opensalt/opensalt/263#-KzZrGdRcxkJZlJq9BoD:-KzZrGdRcxkJZlJq9BoE:b-f5ykcs) ([raw file](https://github.com/opensalt/opensalt/blob/a657d1aabe705c38aff0e1edf3fd0dfc3edcbf40/src/CftfBundle/Resources/views/DocTree/view.html.twig#L122)):*
> ```Twig
>     <div id="treeSideRight" class="treeSide borderRight">
>         <div class="rightTreeSideLeftInner" >
>             <div class="toggle-container-right"> <i class="fa fa-chevron-circle-left" aria-hidden="true" id="toggleLeft"></i></div></div>
> ```

The second closing div should probably be moved to line up with what it closes.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensalt/opensalt/263#-:-KzZqiI0Jjb2ACcC3nBq:bgc0gcp)*
<!-- Sent from Reviewable.io -->

roverwolf commented:
Reviewed 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[app/Resources/assets/js/lsdoc/index.js, line 424 at r3](https://reviewable.io:443/reviews/opensalt/opensalt/263#-KzdMdjPkJta2THo2Joi:-KzdMdjPkJta2THo2Joj:b-5j32gg) ([raw file](https://github.com/opensalt/opensalt/blob/ab1ba660eccf63e1625887e87de58cf109f6eb0a/app/Resources/assets/js/lsdoc/index.js#L424)):*
> ```JavaScript
> })();
> 
> document.getElementById("toggleRight").onclick = function () {
> ```

Note that this is also being run on the main `/cfdoc/` page (the focus of the `.../lsdoc/index.js` file) which does **not** have the `toggleRight` and `toggleLeft` elements so an error is being thrown stopping the rest of the js file from running (hence the tests failing).

It might be better to put the initialization code into the `.../cftree/` files as a function and called from the `apx.initialize()` function.

Is there a reason we are using `document.getElementById()` here instead of jQuery `$()` since the rest of the code that was added is depending on jQuery anyway?

---

*[src/CftfBundle/Resources/views/DocTree/view.html.twig, line 82 at r3](https://reviewable.io:443/reviews/opensalt/opensalt/263#-KzdJhBezcALfScivxw6:-KzdJhBezcALfScivxw7:b-bxopwr) ([raw file](https://github.com/opensalt/opensalt/blob/ab1ba660eccf63e1625887e87de58cf109f6eb0a/src/CftfBundle/Resources/views/DocTree/view.html.twig#L82)):*
> ```Twig
>                     {% if editorRights  and isDraft %}
>                     <div class="input-group" style="clear:both">
> 					<span id="tree1SectionBulkControls" class="input-group-btn">
> ```

Should use spaces instead of tabs in these 15 lines (and another 10 lines later in the file).

---

*[src/CftfBundle/Resources/views/DocTree/view.html.twig, line 119 at r3](https://reviewable.io:443/reviews/opensalt/opensalt/263#-KzdJMRA071z7UqIvP62:-KzdJMRA071z7UqIvP63:bt8xk9d) ([raw file](https://github.com/opensalt/opensalt/blob/ab1ba660eccf63e1625887e87de58cf109f6eb0a/src/CftfBundle/Resources/views/DocTree/view.html.twig#L119)):*
<details><summary><i alt="summary">Quoted 7 lines of code…</i></summary>
>                            <li>To <b>create statements/items</b> click the â€œAdd New Child Itemâ€� button at right.
>                             </li>
>                             <li>To <b>create â€œcrosswalkâ€� associations</b> between two different documents, start by
>                                 clicking â€œChange Documentâ€� above, then choose a different document to view here.
>                             </li>
>                         {% endif %}
>                         <li>To <b>view existing crosswalk associations</b>, click the â€œAssociation Viewâ€� button
</details>

It looks like the ” chracters got changed to some odd characters by the editor here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensalt/opensalt/263)*
<!-- Sent from Reviewable.io -->

Vishvesh-LM commented:
Dave, reason for not using jQuery is that for other projects we do at ACT, we're asked not to use jQuery. We    now have their approval to use jQuery for OpenSALT but in the meantime, we had to use basic JavaScript. 

Is it ok to keep it for now and then in next sprint, we move this code to jQuery?

roverwolf commented:
Yes, not using jQuery is fine.  I was curious since you used jQuery in the dragging code but not the toggle code.

---

Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/opensalt/opensalt/263#-:-KzycjEH6ArVT8CumK88:b-jxs439)*
<!-- Sent from Reviewable.io -->

anuradhas248 commented:
@roverwolf  - As per the vishvesh's comment, for now I am keeping javascript code as it is. I have handled 'toggleRight and toggleLeft elements missing' error. Also made indentation changes as mentioned in review comment. Can you please review code once from your side. If everything is fine, then would you mind to merge kms-2279 branch in main branch.

roverwolf commented:
Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/CftfBundle/Resources/views/DocTree/view.html.twig, line 119 at r3](https://reviewable.io:443/reviews/opensalt/opensalt/263#-KzdJMRA071z7UqIvP62:-L-2Db4TXKTuB9ULAAgZ:b-v8tv26) ([raw file](https://github.com/opensalt/opensalt/blob/ab1ba660eccf63e1625887e87de58cf109f6eb0a/src/CftfBundle/Resources/views/DocTree/view.html.twig#L119)):*
<details><summary><i>Previously, roverwolf (David Ward) wrote…</i></summary><blockquote>

>                            <li>To <b>create statements/items</b> click the â€œAdd New Child Itemâ€� button at right.
>                             </li>
>                             <li>To <b>create â€œcrosswalkâ€� associations</b> between two different documents, start by
>                                 clicking â€œChange Documentâ€� above, then choose a different document to view here.
>                             </li>
>                         {% endif %}
>                         <li>To <b>view existing crosswalk associations</b>, click the â€œAssociation Viewâ€� button

It looks like the ” chracters got changed to some odd characters by the editor here.
</blockquote></details>

There are still odd characters in this file where your editor changed the previous “ and ” characters.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensalt/opensalt/263)*
<!-- Sent from Reviewable.io -->

anuradhas248 commented:
I have added Nilesh's dactepicker changes for kms-2252 bug. Could you please review those changes with this code?

roverwolf commented:
It is quite difficult to change the year with the datepicker chosen to be used here.

The start year of a framework that is being added based on a state framework will often be several years ago, and having to go back dozens of months seems very awkward.

The datepicker is *not* being applied when editing an existing framework, only for creating a framework.

I'm also wondering if a datepicker should only be added if the browser does not have a built-in datepicker (for consistency with expectations in a browser).

---

Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[src/CftfBundle/Resources/views/DocTree/view.html.twig, line 47 at r5](https://reviewable.io:443/reviews/opensalt/opensalt/263#-L-2MyobgT43vBHzAVqH:-L-2MyobgT43vBHzAVqI:b-6fhv4n) ([raw file](https://github.com/opensalt/opensalt/blob/6def06254c8a58e4f951293ce87ac16e472477b3/src/CftfBundle/Resources/views/DocTree/view.html.twig#L47)):*
>                 {% if isDraft %}
The `isDraft` check seems to have been lost here.

---

*[src/CftfBundle/Resources/views/DocTree/view.html.twig, line 131 at r5](https://reviewable.io:443/reviews/opensalt/opensalt/263#-L-2PwYm1uX-pL0Fjq7E:-L-2PwYm1uX-pL0Fjq7F:byui38d) ([raw file](https://github.com/opensalt/opensalt/blob/1cfee9eac130e0a332058a3a53d4b69666107e5b/src/CftfBundle/Resources/views/DocTree/view.html.twig#L131)):*
>             <div id="noItemsInstructions"

It looks like this line was accidentally duplicated as it is duplicating an HTML id which is not valid.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensalt/opensalt/263#-:-L-2JIeA7rGBCG1Vyio7:b-hfdg3a)*
<!-- Sent from Reviewable.io -->

roverwolf commented:
@anuradhas248 I'll note that I cannot deploy to ACT's systems if that is what you are referring to when asking to deploy on staging.

roverwolf commented:
This PR is for #259

roverwolf commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[app/Resources/assets/js/lsdoc/index.js, line 633 at r4](https://reviewable.io:443/reviews/opensalt/opensalt/263#-L-3dMceNbRc0NKfzHzW:-L-3dMceNbRc0NKfzHzX:b-8xjx4) ([raw file](https://github.com/opensalt/opensalt/blob/3984bac557f6efd269fc0fe5b1b0155e19162aed/app/Resources/assets/js/lsdoc/index.js#L633)):*
> if (dragging)

It seems that `dragging` is never set back to false, so once one drags the **dragbar** any click after that on the page moves it again.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensalt/opensalt/263)*
<!-- Sent from Reviewable.io -->

roverwolf commented:
How about something along the lines of the following?  (fixes issue with `dragging` not getting set back to false, and also make the bar move with the cursor)

```
$('#dragbar').mousedown(function(e){
    e.preventDefault();

    var dragBar = function (e) {
        var maxWidthToAllow=$("#treeView").width()-330;
        if (e.pageX >= maxWidthToAllow) {
            $('#treeSideLeft').css("width", maxWidthToAllow);
        } else {
            $('#treeSideLeft').css("width", e.pageX+2);
        }
        $('#treeSideRight').width($("#treeView").width() - $("#treeSideLeft").width());
    };
    $(document).on('mousemove', dragBar);

    $(document).one('mouseup', function(e){
        $(document).off('mousemove', dragBar);
    });
});
```

**Note**, the `dragbar` should probably be hidden when the `treeLeftSide` is toggled to full width so that it is not displayed on the page (to be consistent with when the right side is full width), or else the code above should check to see if `treeRightSide` is not hidden before adding the mouse event handlers for dragging.

roverwolf commented:
Also, we need to have acceptance tests for the ci system to show that this is working before we would want to merge it into the develop branch.

anuradhas248 commented:
@roverwolf - I have updated code as per your comments. Please verify. Also I have removed date picker code for now, as Nilesh need some time to complete functionality. We would add this functionality in other pull request.

roverwolf commented:
You still have not fixed the quotes that got munged by your editor in the `view.html.twig` file.

We still need at least one automated acceptance test created for the ci system to be able to check that this is working.

---

Reviewed 2 of 2 files at r6.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[app/Resources/assets/sass/application.scss, line 28 at r6](https://reviewable.io:443/reviews/opensalt/opensalt/263#-L-7PWtiSxD8Qe-WIf-W:-L-7PWtjJLLZ7G_kz0-k:bk6cgk2) ([raw file](https://github.com/opensalt/opensalt/blob/341e53452172f7101b558dd37473b983cbe77ad4/app/Resources/assets/sass/application.scss#L28)):*
> @import "../../../../vendor/bower-asset/jquery-ui/themes/smoothness/jquery-ui.css";

This should probably be removed as it was part of the datepicker that was removed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensalt/opensalt/263#-:-L-7R4WoBDRB4v6s2iYS:b-mciqm9)*
<!-- Sent from Reviewable.io -->

anuradhas248 commented:
@roverwolf- would you mind to mention line numbers, where you can find the extra quote. I have compared with old file but not able to find any thing.

roverwolf commented on src/CftfBundle/Resources/views/DocTree/view.html.twig:
```diff
@@ -5,510 +5,625 @@
+                    currently contain any statements/items.
+                    <ul>
+                        {% if editorRights and isDraft %}
+                            <li>To <b>create statements/items</b> click the â€œAdd New Child Itemâ€� button at right.
```
Lines 117 through 123 have the quotes munged to be different characters.

Around `Add New Child Item`, `crosswalk`, `Change Document` and `Association View`

anuradhas248 commented:
@roverwolf - I have removed extra quotes and also added acceptance test cases. Please verify and confirm.

anuradhas248 commented:
@roverwolf : Could you please review the changes?

roverwolf commented:
- You have now removed the quotes entirely instead of just fixing how you munged them, the quotes should be put back to help with the instructions
- You will need to merge in the changes from the develop branch

roverwolf commented:
Reviewed 1 of 1 files at r7.
Review status: 3 of 5 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[src/CftfBundle/Resources/views/DocTree/view.html.twig, line 249 at r8](https://reviewable.io:443/reviews/opensalt/opensalt/263#-L-bTxO0hWMwpJeevqU1:-L-bTxO0hWMwpJeevqU2:b-ipzrni) ([raw file](https://github.com/opensalt/opensalt/blob/6e664fcab6b4a99302cd461ac5b88ed705fba2a6/src/CftfBundle/Resources/views/DocTree/view.html.twig#L249)):*
> <a href="{{ path('export_comment', {'id':lsDocId}) }}" class="btn btn-default btn-sm"
>                                                         data-alt-document-disabled="true">Export Comment
>                                                 </a>

Where is this from?  There is no such route as `export_comment`, so this doesn't work.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensalt/opensalt/263)*
<!-- Sent from Reviewable.io -->

anuradhas248 commented:
@roverwolf - I have resolved the conflict. Also I have discussed all comments with Jorge and he seems everything is fine. Could you please verify from your side.

roverwolf commented:
@anuradhas248 You still have not put the quotes back the way they were before.

roverwolf commented:
Reviewed 2 of 2 files at r8, 2 of 2 files at r9, 2 of 2 files at r10.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/opensalt/opensalt/263)*
<!-- Sent from Reviewable.io -->

roverwolf commented:
Related to #244
